src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs

```diff
@@ -348,7 +343,7 @@ public void Stop()
 
             using (await this.consensusLock.LockAsync(this.nodeLifetime.ApplicationStopping).ConfigureAwait(false))
             {
-                blockValidationContext.RuleContext = new RuleContext(blockValidationContext, this.Validator.ConsensusParams, this.Tip);
+                blockValidationContext.RuleContext = new RuleContext(blockValidationContext, this.Chain.Network.Consensus, this.Tip);
```

Comment:
Maybe here it will make more sense to have Network taken from the CI instead of the Chain?

Reply:
Maybe, I don't know definitively either way, so I'll take whoever's advice on this. Just let me know

Reply:
@dangershony mentioned that `ConsensusLoop` will go anyway, so said to leave it as this for now.